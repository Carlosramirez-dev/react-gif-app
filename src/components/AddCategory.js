import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');

    const handleChange = (e) => {
        //console.log(e.target.value);
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories( cat => [inputValue, ...cat]);
            setinputValue('');
        }
        //console.log('Submit Echo');
       
    }

    return (
            <form onSubmit={handleSubmit}>
                 <input 
                    type="text" 
                    value={inputValue}
                    onChange={handleChange}
                />
            </form>
           
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
