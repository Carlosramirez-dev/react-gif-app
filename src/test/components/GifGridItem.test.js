import { shallow } from 'enzyme/build';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'A title';
    const url = 'https://localhost/a/image.gif';
    const wraper = shallow(<GifGridItem title={title} url={url} />);

    test('debe de mostrar <GifGridItem/> correctamente ', () => {
      
        expect( wraper ).toMatchSnapshot();
    });
    
    test('debe de tener un parrafo con el title ', () => {
        
        const p = wraper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('debe de tener la imagen igual al url y alt de los props ', () => {
        
        const img = wraper.find('img');
        // console.log(img.props('src'));

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener animate_fadeIn ', () => {
        
        const div = wraper.find('div');
        const className = div.prop('className');
        
        // console.log(div.prop('className'));
        expect(className.includes('animate__fadeInLeft')).toBe(true);
    })
    
    
    
    
})