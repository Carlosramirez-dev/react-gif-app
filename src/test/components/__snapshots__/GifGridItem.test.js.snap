// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar <GifGridItem/> correctamente  1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="A title"
    src="https://localhost/a/image.gif"
  />
  <p>
    A title
  </p>
</div>
`;
